orchestrator/utils/search_query.py

```diff
@@ -255,6 +257,29 @@ def parse(self) -> Node:
 
 
 class TSQueryVisitor:
+
+    _glue_chars = re.compile(r"[-_@#$%^&]")
+
+    @staticmethod
+    def _split_term(term: str) -> list[str]:
+        """Workaround for the way Postgres <14 parses text with to_tsquery.
+
+        For Postgres <14, we issue a database query to parse the text for us in a way that is compatible with to_tsvector. This is the same behavior as Postgres >=14.
+        """
+
+        # TODO: Remove this workaround when support for Postgres <14 is dropped
+
+        if get_postgres_version() < 14:
```

Comment:
Maybe store the version in a global variable or contextvar, to reduce the overhead to the first query that an API instance processes? 

We won't ever do a major upgrade on the database while keeping the app running, so there's no concern of this value going stale. And it's temporary code anyway.

Reply:
`get_postgres_version` is memoized.